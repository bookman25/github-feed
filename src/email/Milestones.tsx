import * as React from 'react';

export interface IMilestones {
	milestones: IMilestone[];
}

export interface IMilestone {
	due_on?: string;
	html_url?: string;
	open_issues?: number;
	title: string;
}

export function Milestones({ milestones }: IMilestones) {
	if (!milestones || milestones.length === 0) {
		return null;
	}

	return (
		<p>
			<strong>Milestones: </strong>
			{ milestones.map((milestone, m) => (
				<span key={ m }>
					{ m !== 0 && ' | ' }
					<i><a href={ milestone.html_url }>{ milestone.title }</a></i> (open: { milestone.open_issues }
					{ milestone.due_on && `, due: ${milestone.due_on}` })
			</span>
			)) }
		</p>
	);
}