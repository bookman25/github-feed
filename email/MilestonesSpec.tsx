jest.unmock('./Milestones');

import * as React from 'react';
import { Milestones, IMilestone } from './Milestones';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('Milestones', () => {
	const milestone1: IMilestone = {
		title: 'Milestone 1',
		html_url: 'url',
		open_issues: 1,
	};

	it('with no milestones', () => {
		const view = shallow(<Milestones milestones={ [] } />);
		expect(view.type()).toBeNull();
	});

	it('multiple milestones', () => {
		const milestones: IMilestone[] = [
			milestone1,
			Object.assign({}, milestone1, {
				title: 'Milestone 2',
				html_url: 'url2',
			}),
		];
		const view = shallow(<Milestones milestones={ milestones } />);
		expect(shallowToJson(view)).toMatchSnapshot();
	});

	it('without due date', () => {
		const view = shallow(<Milestones milestones={ [milestone1] } />);
		expect(shallowToJson(view)).toMatchSnapshot();
	});

	it('with due date', () => {
		const milestone = Object.assign({}, milestone1, {
			due_on: 'due date',
		});
		const view = shallow(<Milestones milestones={ [milestone] } />);
		expect(shallowToJson(view)).toMatchSnapshot();
	});
});