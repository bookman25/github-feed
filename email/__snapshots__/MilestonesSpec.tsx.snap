exports[`Milestones multiple milestones 1`] = `
<p>
  <strong>
    Milestones: 
  </strong>
  <span>
    <i>
      <a
        href="url">
        Milestone 1
      </a>
    </i>
     (open: 
    1
    )
  </span>
  <span>
     | 
    <i>
      <a
        href="url2">
        Milestone 2
      </a>
    </i>
     (open: 
    1
    )
  </span>
</p>
`;

exports[`Milestones with due date 1`] = `
<p>
  <strong>
    Milestones: 
  </strong>
  <span>
    <i>
      <a
        href="url">
        Milestone 1
      </a>
    </i>
     (open: 
    1
    , due: due date
    )
  </span>
</p>
`;

exports[`Milestones without due date 1`] = `
<p>
  <strong>
    Milestones: 
  </strong>
  <span>
    <i>
      <a
        href="url">
        Milestone 1
      </a>
    </i>
     (open: 
    1
    )
  </span>
</p>
`;
